From bb3aeacdaf7f824d3e1eb382e930ea8f0a1c6962 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Sun, 12 Jul 2026 13:44:27 +0300
Subject: [PATCH] Platform: Don't assert on Variable store check integrity 
 failure

This change ends up both standalonemm_jetson.pkg used on the AGX Thor.

We continue to avoid the reset because In this configuration the rollback tests passed
on the AGX Orin when migrating to Jetpack 5 to Jetpack 6, if the store
integrity check failed, and it allowed to continue the OS update.

We thus continue to apply this change to the AGX Thor to avoid
the need of re-flashing devices in the field if the var store
gets accidentally corrupted.

Upstream-status: Inappropriate [Configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io.
---
 Platform/NVIDIA/StandaloneMmJetson/StandaloneMmJetson.dsc.inc | 2 +-
 .../NVIDIA/Drivers/DefaultVariableDxe/DefaultVariableDxe.c    | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Platform/NVIDIA/StandaloneMmJetson/StandaloneMmJetson.dsc.inc b/Platform/NVIDIA/StandaloneMmJetson/StandaloneMmJetson.dsc.inc
index 18fd2bcf..03a92343 100644
--- a/Platform/NVIDIA/StandaloneMmJetson/StandaloneMmJetson.dsc.inc
+++ b/Platform/NVIDIA/StandaloneMmJetson/StandaloneMmJetson.dsc.inc
@@ -45,7 +45,7 @@
 
 [PcdsFeatureFlag]
  gNVIDIATokenSpaceGuid.PcdVarStoreIntegritySupported|TRUE
- gNVIDIATokenSpaceGuid.PcdAssertOnVarStoreIntegrityCheckFail|TRUE
+ gNVIDIATokenSpaceGuid.PcdAssertOnVarStoreIntegrityCheckFail|FALSE
 
 [PcdsPatchableInModule]
 
diff --git a/Silicon/NVIDIA/Drivers/DefaultVariableDxe/DefaultVariableDxe.c b/Silicon/NVIDIA/Drivers/DefaultVariableDxe/DefaultVariableDxe.c
index 41449ca6..0c1caf75 100644
--- a/Silicon/NVIDIA/Drivers/DefaultVariableDxe/DefaultVariableDxe.c
+++ b/Silicon/NVIDIA/Drivers/DefaultVariableDxe/DefaultVariableDxe.c
@@ -75,7 +75,7 @@ CheckVarStoreState (
     DEBUG ((DEBUG_ERROR, "%a: %s not found %r\n", __FUNCTION__, VARINT_CHECK_FAILED, Status));
   } else {
     if (VarStoreCheckFailed == 1) {
-      DEBUG ((DEBUG_ERROR, "Rebooting the System to re-init Var Store\n"));
+      DEBUG ((DEBUG_ERROR, "NOT Rebooting the System to re-init Var Store\n"));
       Status = gRT->SetVariable (
                       VARINT_CHECK_FAILED,
                       &gEfiGlobalVariableGuid,
@@ -87,7 +87,7 @@ CheckVarStoreState (
         DEBUG ((DEBUG_ERROR, "%aFailed to Delete %s %r\n", __FUNCTION__, VARINT_CHECK_FAILED, Status));
       }
 
-      ResetCold ();
+      //ResetCold ();
     }
   }
 }
-- 
2.51.0

